node/src/components/consensus/highway_core/state.rs

```diff
@@ -688,3 +693,23 @@ fn log2(x: u64) -> u32 {
 fn leader_prng(upper: u64, seed: u64) -> u64 {
     ChaCha8Rng::seed_from_u64(seed).gen_range(0, upper) + 1
 }
+
+/// Returns whether `vote` cites a new vote from `vidx` in the last panorama.
+/// i.e. whether previous vote from creator of `vhash` cites different vote by `vidx`.
+///
+/// NOTE: Returns `false` if `vidx` is faulty or hasn't produced any messages according to the
+/// creator of `vhash`.
+pub(crate) fn new_hash_obs<C: Context>(
```

Comment:
Why not make this a method of `State`? (Or one of `Vote`?)

Reply:
I didn't feel like it should be a method of `State`, but `Vote` sounds better :+1:

Reply:
Done. Moved to `Vote`.